### ISSUE TITLE
Enable EKS Pod Identity Association 

### ISSUE THREAD

**Issue by IdoOzeri** [kubernetes/autoscaler#6461]
I've moved from using IRSA with Cluster Autoscaler, to Pod Identity Association. 
I used to specify the ARN of the IAM role in the autoscaler's Kubernetes Service Account and everything worked perfectly. 

Now that I've switched to Pod Identity Association, I've removed the role ARN annotation from the values file, so it looks like this:

values.yaml
```
awsRegion: eu-central-1
namespace: kube-system

autoDiscovery:
  clusterName: eks-dev

rbac:
  serviceAccount:
    name: "cluster-autoscaler"

deployment:
  annotations:
    "cluster-autoscaler.kubernetes.io/safe-to-evict": "false"

extraArgs:
  "skip-nodes-with-system-pods" : "false"
  "balance-similar-node-groups" : "true"
```

However, the autoscaler pod is in CrashLoopBackoff with:
```
I0121 23:49:31.768485       1 auto_scaling_groups.go:393] Regenerating instance to ASG map for ASG names: []
I0121 23:49:31.768510       1 auto_scaling_groups.go:400] Regenerating instance to ASG map for ASG tags: map[k8s.io/cluster-autoscaler/eks-dev: k8s.io/cluster-autoscaler/enabled:]
E0121 23:49:31.768621       1 aws_manager.go:126] Failed to regenerate ASG cache: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
F0121 23:49:31.768650       1 aws_cloud_provider.go:447] Failed to create AWS Manager: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
```

I've verified the service account can indeed perform all the `autoscaling/ec2` operations by creating a dummy Ubuntu pod with the service account mounted and aws-cli. I was able to run all the commands freely. 

I figured the move to Pod Identity would be transparent for the autoscaler. 

Is it being supported? 
Should it matter to the autoscaler?
Has anyone got it to work? 

I'm stuck on it for a couple of days now, could really appreciate any feedback, thanks!

**Comment by AleCo3lho** [comment]
Hey, what is the version of cluster autoscaler that you are using?

Try it with the latest version and let me know if it works.

For me that did the trick.

**Comment by TreeKat71** [comment]
It works for me after changing to `v1.29.0`

**Comment by Shubham82** [comment]
As per the following comment: https://github.com/kubernetes/autoscaler/issues/6392#issuecomment-1909751777
it is working with CA 1.29
@IdoOzeri Could you please take a look?

**Comment by Ido-Ozeri** [comment]
> As per the following comment: https://github.com/kubernetes/autoscaler/issues/6392#issuecomment-1909751777
> it is working with CA 1.29
> @IdoOzeri Could you please take a look?

It is working. Thanks a bunch 👍

**Comment by Shubham82** [comment]
Thanks, @IdoOzeri for the confirmation.

**Comment by Shubham82** [comment]
closing this issue, as it is resolved in CA 1.29
